docs/format_howtos.py

```diff
@@ -0,0 +1,78 @@
+"""
```

Comment:
How difficult is it to incorporate this into sphinx? Ideally we would load this code as a sphinx plugin and use something like `howto:: [name]` in the rst files.

Reply:
I was thinking about that as well. Honestly I don't know, but the approach in this CL has the added benefit that it's easy to debug/test the format_howto script. If it were entirely in Sphinx I could see that being harder. This version is more like "unix standalong executables that combine via pipes".

So, given that this approach works, doesn't have obvious flaws and allows for testing/debugging, I'd like to err on the side of "let's keep it the way it is".